### PR TITLE
fix(pet-82): replace HelpTip innerHTML with structured markup builder

### PR DESCRIPTION
**Pull request by ziomancer**
## Summary
- Replaces the only variable-fed `innerHTML` sink in the console frontend (`Pet.HelpTip`) with `Pet.richText`, a zero-dependency markup builder that parses a restricted tag subset (`<b>`, `<code>`, `<em>`) into real DOM nodes and escapes everything else to text.
- No XSS sink can be smuggled through a future dynamic tooltip — only `b`/`code`/`em` bare tags can ever become elements; no attribute, event handler, or foreign element is creatable.
- Adds the repo's first JS test (Node built-in runner, zero npm deps) plus a CI `js-test` job so the helper is guarded on every PR.

## Layered defense
`Pet.richText` never constructs an HTML string and never touches `innerHTML`/`DOMParser`. The allow-list regex `^<(\/?)(b|code|em)>$/i` is anchored, so any tag with attributes, whitespace, an embedded `>`, or a slash (`<b/>`) fails the match and degrades to a literal text node. `createElement` is only ever called with the captured tag name (`b`/`code`/`em`, lowercased) — an attacker cannot reach attribute or handler creation. All 6 current callers pass static `<b>`/`<code>` literals and render unchanged.

## Files changed

| File | Change |
|------|--------|
| `petasos/console/static/petasos.js` | Adds `Pet.richText`; `HelpTip` now `appendChild(Pet.richText(html))` instead of `innerHTML = html` |
| `tests/js/richtext.test.mjs` | New — 20-case unit test; minimal DOM shim; loads the real `petasos.js` via `node:vm` |
| `.github/workflows/ci.yml` | New `js-test` job (setup-node, pinned Node 22, no npm install) |

## Test plan
- [x] `node --test tests/js/richtext.test.mjs` — 20/20 pass (full output: docs/specs/TODO/PET-82.test-output.txt)
- [x] Allowed (`<b>`/`<code>`/`<em>`), nested, case-insensitive, mis-nested, unclosed/unbalanced, truncated, whitespace, astral, and number-coercion cases
- [x] Adversarial security sweep (`<script>`, ``, `<svg/onload>`, `<iframe>`, `<a href=javascript:>`, `<b title=">">`, `<b/>`, newline-in-tag) — no element ever smuggled
- [x] Structural equivalence for all 6 current HelpTip caller strings + `&`/stray-angle-bracket tripwire
- [ ] Manual visual spot-check of the running console tooltips (Observability, Scan Playground, About) — post-merge

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Help tips now render safe rich text (bold, code, emphasis) while preventing unsafe markup.

* **Tests**
  * Added comprehensive unit tests covering parsing edge cases, nesting, case-insensitivity, text coercion, and security/XSS checks.

* **Chores**
  * Added a CI job to run the new JavaScript tests in the pipeline.

* **Documentation**
  * Added a recorded test run output documenting passing results and behavior observations.

### CODE REVIEW

File: petasos/console/static/petasos.js [hunk 1]
```diff
@@ -104,11 +104,57 @@
...
+          if (m) {
+            flush();
+            if (m[1] === "/") {
+              if (stack.length > 1) stack.pop();
+            } else {
+              var el = document.createElement(m[2].toLowerCase());
+              stack[stack.length - 1].appendChild(el);
+              stack.push(el);
+            }
+            i = close + 1;
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/static/petasos.js]: Resolved in bf03f5c — closing tags now pop only when they match the current open element's tagName; a mismatched closer is treated as literal text, so mis-nested/unbalanced structure is no longer silently suppressed. Added test #9 (`<b><em>x</b>`) plus updated #8/#11.
